tools/rows_picker_gen.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+func main() {
+	var err error
+	fn := flag.String("o", "", "output file")
+	flag.Parse()
+
+	out := os.Stdout
+	if *fn != "" {
+		out, err = os.Create(*fn)
+		if err != nil {
+			log.Fatalf("could not create file %q, %v", *fn, err)
+		}
+	}
+	defer out.Close()
+
+	intfs := []string{
+		"NextResultSet",
+		"ColumnTypeDatabaseTypeName",
+		"ColumnTypeLength",
+		"ColumnTypeNullable",
+		"ColumnTypePrecisionScale",
+		"ColumnTypeScanType",
+	}
+
+	fmt.Fprintln(out, "package sqlmw")
+
+	fmt.Fprintln(out, "")
+	fmt.Fprintln(out, "import (")
+	fmt.Fprintln(out, "\t\"context\"")
+	fmt.Fprintln(out, "\t\"database/sql/driver\"")
+	fmt.Fprintln(out, ")")
+
+	fmt.Fprintln(out, "")
+	genConst(out, intfs)
+
+	fmt.Fprintln(out, "")
+	genPickerTable(out, intfs)
+
+	fmt.Fprintln(out, "")
+	genWrapRows(out, intfs)
+}
+
+/*
+const (
+	rowsNextResultSet = 1 << iota
+	rowsColumnTypeDatabaseTypeName
+	rowsColumnTypeLength
+	rowsColumnTypeNullable
+	rowsColumnTypePrecisionScale
+	rowsColumnTypeScanType
+)
```

Comment:
remove?

Reply:
> Thank you for this contribution! Can we add a few tests that exercise the new functionality and prove its correctness?

yes, I think so. The Go mysql, pgx and sqlite3 drivers expose different combiantions of these, so I was thinking of having mock drivers that match what they use. 
  It'll take me a while to get the tests sorted, didn't want to start the effort until I was sure you'd be interested in the merge.